Content.Client/CriminalRecords/GeneralCriminalRecordConsoleWindow.xaml.cs

```diff
@@ -0,0 +1,289 @@
+            if (_isPopulating || RecordListing[args.ItemIndex].Metadata is not StationRecordKey cast)
+            {
+                return;
+            }
+
+            OnKeySelected?.Invoke(cast);
+        };
+
+        RecordListing.OnItemDeselected += _ =>
+        {
+            if (!_isPopulating)
+                OnKeySelected?.Invoke(null);
+        };
+
+        StationRecordsFilterType.OnItemSelected += eventArgs =>
+        {
+            var type = (GeneralStationRecordFilterType)eventArgs.Id;
+
+            if (_currentFilterType != type)
+            {
+                _currentFilterType = type;
+                FilterListingOfRecords();
+            }
+        };
+
+        StationRecordsFiltersValue.OnTextEntered += args =>
+        {
+            FilterListingOfRecords(args.Text);
+        };
+
+        StationRecordsFilters.OnPressed += _ =>
+        {
+            FilterListingOfRecords(StationRecordsFiltersValue.Text);
+        };
+
+        StationRecordsFiltersReset.OnPressed += _ =>
+        {
+            StationRecordsFiltersValue.Text = "";
+            FilterListingOfRecords();
+        };
+
+        ArrestButton.OnPressed += e => OnArrestButtonPressed?.Invoke(e, ReasonLineEdit.Text, _recordName);
+
+        StatusOptionButton.OnItemSelected += args =>
+        {
+            args.Button.SelectId(args.Id);
+            OnStatusOptionButtonSelected?.Invoke(args, (SecurityStatus) StatusOptionButton.GetItemMetadata(args.Id)!,
+                ReasonLineEdit.Text, _recordName);
+        };
+
+    }
+
+    public void UpdateState(GeneralCriminalRecordConsoleState state)
+    {
+        if (state.Filter != null)
+        {
+            if (state.Filter.Type != _currentFilterType)
+            {
+                _currentFilterType = state.Filter.Type;
+            }
+
+            if (state.Filter.Value != StationRecordsFiltersValue.Text)
+            {
+                StationRecordsFiltersValue.Text = state.Filter.Value;
+            }
+        }
+
+        StationRecordsFilterType.SelectId((int)_currentFilterType);
+
+        if (state.RecordListing == null)
+        {
+            RecordListingStatus.Visible = true;
+            RecordListing.Visible = false;
+            RecordListingStatus.Text = Loc.GetString("general-criminal-record-console-empty-state");
+            RecordContainer.Visible = false;
+            RecordContainerStatus.Visible = false;
+            return;
+        }
+
+        RecordListingStatus.Visible = false;
+        RecordListing.Visible = true;
+        RecordContainer.Visible = true;
+
+        PopulateRecordListing(state.RecordListing!, state.SelectedKey);
+
+        RecordContainerStatus.Visible = state.CriminalRecord == null;
+
+        ArrestButton.Text = state.CriminalRecord?.Status == SecurityStatus.Detained ? Loc.GetString("general-criminal-record-release-button") : Loc.GetString("general-criminal-record-arrest-button");
+
+        ReasonLineEdit.Visible = state.SelectedKey != null;
```

Comment:
don't copy and paste this, just check it once - in fact, you can just nest all of these into another container and then just make that single one invisible

Reply:
I moved all those into a new `BoxContainer` and tried to disable its visibility, but it never worked. I tried it twice, nothing. Either I'm dumb or something doesn't work as wanted. So I'll resolve it for now.